machine_learning_hep/analysis/analyzer_jet.py

```diff
@@ -391,15 +405,18 @@ def fit(self): # pylint: disable=too-many-branches, too-many-locals
                 histomass_reb.Copy(histomass_reb_f)
                 fitter = AliHFInvMassFitter(histomass_reb_f, self.p_massmin[ipt], \
                     self.p_massmax[ipt], self.p_bkgfunc[ipt], self.p_sgnfunc[ipt])
-                fitter.SetInitialGaussianSigma(fittermc.GetParameter(2))
-                fitter.SetInitialGaussianMean(fittermc.GetParameter(1))
-                if self.p_set_initial_sigma[ipt] is True:
-                    if self.p_set_fix_sigma[ipt] is True:
+                if sgn_mc:
+                    if self.p_set_initial_sigma[ipt]:
+                        fitter.SetInitialGaussianSigma(self.p_sigmaarray[ipt])
+                    else:
+                        fitter.SetInitialGaussianSigma(sigma_mc)
+                    fitter.SetInitialGaussianMean(mean_mc)
+                    if self.p_set_fix_sigma[ipt]:
                         print("****************Fix sigma to sigma array values*****************")
                         fitter.SetFixGaussianSigma(self.p_sigmaarray[ipt])
-                    else:
+                    if self.p_set_mc_sigma[ipt]:
                         print("****************Fix sigma to MC sigma values*****************")
-                        fitter.SetFixGaussianSigma(fittermc.GetParameter(2))
+                        fitter.SetFixGaussianSigma(sigma_mc)
```

Comment:
i find this whole section a bit confusing. Isnt all you need the following?

set_sigma = 0
if self.p_set_initial_sigma[ipt]:
  set_sigma = self.p_sigmaarray[ipt]
else:
   if sgn_mc:
       set_sigma = fittermc.GetParameter(2)

if p_set_fix_sigma[ipt]:
   fitter.SetFixGaussianSigma(set_sigma)

would this work? it then makes the fields in the database very simple to understand and we dont need this extra self.p_set_mc_sigma[ipt]:

Reply:
i put a similar comment before. It has a thumbs up but I dont see this change